lib/iris/tests/unit/util/test_array_equal.py

```diff
@@ -54,6 +54,18 @@ def test_masked_is_different(self):
         array_b = ma.masked_array([1, 2, 2], mask=[0, 0, 1])
         self.assertFalse(array_equal(array_a, array_b))
```

Comment:
Sorry I just spotted another thing : I'm not sure this case is now adding anything useful, since `tset_masked_isnt_unmasked` is already ensuring masks **_must_** be equal (even when all the data is equal).

So, unless I'm missing something, you could just remove this testcase ?

Reply:
The one case which this covers but other tests doesn't is when precisely:

- Both cubes have masks.
- Those cubes are defferent.

`test_masked_isnt_unmasked` tests the case where a mask on one cube is different to the mask which is implied on another cube _without_ an explicit mask. There's maybe a case that this justifies another test even if it's a very narrow case. I'm happy to remove this if you think it's overkill though.

Reply:
OK, I can see we're making the masks explicit in this case.